### PR TITLE
refactor assets, assets importer, and fix Windows paths

### PR DESCRIPTION
**Pull request by eoneill**
This is a substantial refactor.

This pull request contains the following fixes / changes:
- `@import "assets"` within a module does not currently work
  - This was addressed in part by #123
- The root directory is always treated as an "app", even if the root directory is actually a module (`isEyeglassModule === true`).
  - This leads to inconsistencies in assets in the current project vs when the project is consumed as a dependency.
  - See https://github.com/sass-eyeglass/eyeglass/issues/79#issuecomment-173171567
- When using `httpRoot` with `relativeTo`, incorrect asset paths are generated.
- Windows paths separators and delimiters are used inconsistently throughout eyeglass. This takes the first step towards addressing this by introducing a new set of uri utility methods.
  - See #128
- This significantly reorganizes the project structure for better maintainability.

```
lib
├── assets
│   ├── Assets.js
│   ├── AssetsCollection.js
│   └── AssetsSource.js
├── eyeglass-exports.js
├── functions
│   ├── asset-uri.js
│   ├── index.js
│   └── version.js
├── importers
│   ├── AssetImporter.js
│   ├── ImportUtilities.js
│   └── ModuleImporter.js
├── index.js
├── modules
│   ├── EyeglassModule.js
│   ├── EyeglassModules.js
│   └── ModuleFunctions.js
└── util
    ├── NameExpander.js
    ├── Options.js
    ├── debug.js
    ├── deprecator.js
    ├── files.js
    ├── package.js
    ├── resolve.js
    ├── semverChecker.js
    ├── strings.js
    ├── sync.js
    └── uri.js
```

To see individual commit history, please see: https://github.com/eoneill/eyeglass/commits/assets-refactor

### CODE REVIEW

File: lib/util/NameExpander.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ function NameExpander(uri) {
   * @param   {String} location - the location path the expand the URI against
   */
 NameExpander.prototype.addLocation = function(location) {
-  if (location && location !== "stdin" && fileUtils.existsSync(location)) {
+  if (location && location !== "stdin") {
```
Comment (eoneill) [inline lib/util/NameExpander.js]: TODO - I need to add a test for this

**Comment by ryuran** [comment]
Some news about this pull request ?

**Comment by codegecko** [comment]
An update would be appreciated, trying to use bootstrap-sass with Eyeglass on Windows still isn't working and I really need it to (W10 x64 dev here), and trying to check out the git repo with the specific amends fails cause eyeglass's version checking doesn't recognise npm git-based dependencies.

**Comment by Snugug** [comment]
I've run the fork at it's working as expected in Windows. As per @eoneill's instructions, this is what I did:

* Changed my Eyeglass source in `package.json` to this fork/branch (`eoneill/eyeglass#assets-refactor-cleaner`)
* Added `process.env.EYEGLASS_NORMALIZE_PATHS = true;` to my Node file where Eyeglass is run from

After doing those two things, paths on Windows work again.

We've got this working for a bunch of Windows machines; it'd be great if this could make it in so we don't need to point to the fork.

**Comment by chriseppstein** [comment]
Closing. This landed in a different PR.

**Pull request closed without merging**